google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/impl/PubSubApi.scala

```diff
@@ -140,7 +140,15 @@ private[pubsub] trait PubSubApi {
     for {
       request <- Marshal((HttpMethods.POST, uri, request)).to[HttpRequest]
       response <- doRequest(request, maybeAccessToken)
-      pullResponse <- Unmarshal(response).to[PullResponse]
+      pullResponse <- response.status match {
+        case StatusCodes.Success(_) => Unmarshal(response).to[PullResponse]
+        case status =>
+          Unmarshal(response)
+            .to[String]
+            .map { entity =>
+              throw new RuntimeException(s"Unexpected pull response. Code: [$status]. Entity: [$entity]")
+            }
+      }
```

Comment:
I'm using same pattern in pull, acknowledge and publish.
I'm matching for every success status code. If it's not successful, then trying to read error as String to add it to an exception. It works for response with no entity, returning empty string (checked in added spec for acknowledge).